java-frontend/src/main/java/org/sonar/java/resolve/JavaType.java

```diff
@@ -371,9 +365,30 @@ public TypeVariableJavaType(JavaSymbol.TypeVariableJavaSymbol symbol) {
     public JavaType erasure() {
       return bounds.get(0).erasure();
     }
+
+    public List<JavaType> bounds() {
+      return bounds;
+    }
+
+    @Override
+    public boolean isSubtypeOf(Type superType) {
+      JavaType supType = (JavaType) superType;
+      if (supType.isTagged(WILDCARD)) {
+        return ((WildCardType) supType).isSubtypeOfBound(this);
+      }
+      if(supType == this) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)